### PR TITLE
reflect current design choice on readme

### LINKED ISSUE #15: rewrite README
rewrite REAME by reflecting current implementation and design

### PR DESCRIPTION
**Pull request by deep-diver**
Closes #15 

Note Evaluation and Synthetic data generation sections are marked as WIP. These will be updated after writing separate CLIs for them.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,43 +1,58 @@
 # LLMOps Pipeline
 
-This project showcase LLMOps pipeline that fine-tune a small sized LLM model to prepare the outage of the service type LLM. For this project, we choose [Gemini 1.0 Pro](https://deepmind.google/technologies/gemini/) for service type LLM and [Gemma](https://blog.google/technology/developers/gemma-open-models/) 2B/7B for small sized LLM model.
+This project showcase LLMOps pipeline that fine-tune a small size LLM model to prepare the outage of the service LLM. For this project, we choose [Gemini 1.0 Pro](https://deepmind.google/technologies/gemini/) for service type LLM and [Gemma](https://blog.google/technology/developers/gemma-open-models/) 2B/7B for small sized LLM model.
```
Comment (sayakpaul) [inline README.md:3]:
```suggestion
This project showcases LLMOps pipeline that fine-tunes a small sized LLM model to prepare the outage of the service LLM. For this project, we choose [Gemini 1.0 Pro](https://deepmind.google/technologies/gemini/) for service type LLM and [Gemma](https://blog.google/technology/developers/gemma-open-models/) 2B/7B for small-sized LLM.
```

What's an outage? Also, we should make it clear how the small-sized LLM is leveraged and how the service LLM is leveraged.

File: README.md [hunk 2]
```diff
@@ -1,43 +1,58 @@
...
 
 For this project, the following tech stacks are chosen:
-- [TensorFlow Extended](https://www.tensorflow.org/tfx?hl=ko) (TFX) for building Machine Learning pipeline.
-- [KerasNLP](https://keras.io/keras_nlp/) and [Keras](https://keras.io/) in general to fine-tune Gemma 2B/7B.
-- [Gemini API](https://cloud.google.com/vertex-ai/docs/generative-ai/model-reference/gemini) on Google Cloud/Vertex AI.
+- Hugging Face open source ecosystem ([transformers](https://github.com/huggingface/transformers), [peft](https://github.com/huggingface/peft), [alignment-handbook](https://github.com/huggingface/alignment-handbook), [Hub](https://huggingface.co/docs/hub/en/index))
```
Comment (sayakpaul) [inline README.md]:
```suggestion
- Hugging Face open source ecosystem ([`transformers`](https://github.com/huggingface/transformers), [`peft`](https://github.com/huggingface/peft), [`alignment-handbook`](https://github.com/huggingface/alignment-handbook), [`huggingface_hub`](https://huggingface.co/docs/hub/en/index))
```

File: README.md [hunk 3]
```diff
@@ -1,43 +1,58 @@
...
+- [Gemini API](https://cloud.google.com/vertex-ai/docs/generative-ai/model-reference/gemini) on [Google AI Studio](https://ai.google.dev/).
```
Comment (sayakpaul) [inline README.md]: I think we are consuming the API directly. No need to mention Studio I guess.

File: README.md [hunk 4]
```diff
@@ -1,43 +1,58 @@
...
 
-> When we finish up the project, there is possibility that we could add [Gemini API](https://ai.google.dev/) from Google AI Studio instead of Vertex AI. Also, we could showcase how to fine-tune Mistral-7B instead of Gemma models. We will see.
+## Motivation
 
-## Overview
-
-We assume that small sized LLM could show comparable performance to that of service type LLM on a certain task, and this project is to prove such possibility. Furthermore, this project shows how to smoothly migrate from service type LLM to small sized LLM when 
+We assume that small size LLM could show comparable performance to that of service type LLM on a specific task, and this project is to prove such possibility. Furthermore, this project shows how to smoothly migrate from service type LLM to small sized LLM when
```
Comment (sayakpaul) [inline README.md]:
```suggestion
We assume that a small LLM could show comparable performance to that of a service-type LLM on a specific task, and this project tries to showcase such a possibility in a practically grounded manner. Furthermore, this project shows how to smoothly migrate from service-type LLM to small LLM when 
```

File: README.md [hunk 5]
```diff
@@ -1,43 +1,58 @@
...
-- [ ] Notebook to define coverage dataset.
-- [ ] Notebook to define `SynthDataGen` component's functionality.
-- [ ] Notebook to define `LLMEvaluator` component's functionality.
-- [ ] Implement `SynthDataGen` and `LLMEvaluator` as custom TFX component.
-- [ ] Notebook to define a portion of TFX pipeline with the custom component to fine-tune Gemma 2B/7B models.
-- [ ] Implement a whole TFX pipeline including automatically exporting models and application to Hugging Face Hub.
\ No newline at end of file
+## Overview
+
+This project comes with the toolset of batch inference, evaluation, and synthetic data generation. Each tool can be run independently, but they could be hooked up to form a pipeline. It is your choice how to use them.
```
Comment (sayakpaul) [inline README.md]:
```suggestion
This project comes with the toolset of batch inference, evaluation, and synthetic data generation. Each tool can be run independently, but they could be hooked up to form a pipeline. It's on the end user to figure out the best way to collate these together. 
```

File: README.md [hunk 6]
```diff
@@ -1,43 +1,58 @@
...
+
+The prerequisite to run these toolset is to have a dataset consisting of desired `(prompt, response)` pairs. The exact format of the dataset could be found [here](https://huggingface.co/datasets/sayakpaul/no_robots_only_coding). The `prompt` is the input to the small size LLM to generate output. Then, `prompt`, `response`, and the `generated output` are going to be used to evaluate the fine-tuned small size LLM. The main idea is to make small size LLM to output as much as similar to the given response.
+
+### Batch inference
```
Comment (sayakpaul) [inline README.md:35]: Before this there should be a section on fine-tuning.

File: README.md [hunk 7]
```diff
@@ -1,43 +1,58 @@
...
+
+| column names | instructions |  target_responses |  candidate_responses  | model_id  |  model_sha |
+|---|---|---|---|---|---|
+| descriptions | the input prompts | desired outputs |  model generated outputs  |  model id that generated outputs  |  the version of the model |
+
+### Evaluation
+
+Evaluation evaluates the generated text from fine-tuned LLM with the help of service LLM. The evaluation criteria is the similarity and quality by comparing to the given desired outputs.
+
+(Instruction WIP)
```
Comment (sayakpaul) [inline README.md:61]: While they're WIP it could make sense to redirect the users to the notebooks and scripts related to them.

File: README.md [hunk 8]
```diff
@@ -1,43 +1,58 @@
...
+
+### Synthetic data generation
+
+Synthetic data generation generates similar data to the ones used to fine-tune the LLM. This could be performed based on the evaluation results. For instance, if you are not satisfied with the evaluation results, and if you think the training dataset is not large enough, you can create more of the similar data to boost the performance of the LLM.
+
+(Instruction WIP)
+
+## Acknowledgments
+
+This is a project built during the Gemma/Gemini sprints held by Google's ML Developer Programs team. We are thankful to be granted good amount of GCP credits to finish up this project.
```
Comment (sayakpaul) [inline README.md]:
```suggestion
This is a project built during the Gemma/Gemini sprints held by Google's ML Developer Programs team. We are thankful to be granted good amount of GCP credits to finish up this project. Thanks to Hugging Face for providing Sayak with resources to run some fine-tuning experiments. 
```

**Review by sayakpaul — COMMENTED** [review]
Nice start!

**Comment by deep-diver** [comment]
somewhat addressed your comments. used different word, failure instead of outage. 

My English is poor, but I will try my best!

**Comment by sayakpaul** [comment]
No worries. English isn't my native either. README looks good to me 👌

**Pull request merged**